lib/grape/http/headers.rb

```diff
@@ -26,6 +26,10 @@ module Headers
       HTTP_ACCEPT            = 'HTTP_ACCEPT'
 
       FORMAT                 = 'format'
+
+      def self.find_supported_method(route_method)
```

Comment:
The name of the method is `find supported method`, and you are returning unsupported methods from it. I think this method should either `.detect` or return `nil` and the caller should `|| ...`. 

Next, I think we should turn `Grape::Http::Headers::SUPPORTED_METHODS` into a case-insensitive hash and lookup with `Grape::Http::Headers::SUPPORTED_METHODS[method]`, which will be a lot more efficient than `detect`.